tests/unit/slider/slider_options.js

```diff
@@ -143,6 +143,34 @@ test( "orientation", function( assert ) {
 
 	element.slider( "destroy" );
 
+	element = $( "#slider1" ).slider({
+		range: "max",
+		min: 0,
+		max: 10
+	});
+
+	element.slider( "option", "value", 5 );
+	element.slider( "option", "orientation", "vertical" );
+	equal( element.width(), element.find( ".ui-slider-range " ).width(),
+		"range should fill all horizontal space after changing orientation to vertical" );
+
+	element.slider( "option", "orientation", "horizontal" );
+	equal( element.height(), element.find( ".ui-slider-range " ).height(),
+		"range should fill all vertical space after changing orientation to horizontal" );
+
+	element.slider( "destroy" );
+
+	element = $( "#slider1" ).slider({
+		range: true,
+		min: 0,
+		max: 100
+	});
+	element.slider( "option", { values: [ 60, 70 ] } );
+	element.slider( "option", "orientation", "vertical" );
+	equal( element.find( ".ui-slider-range " ).position().left, 0,
```

Comment:
Test the position prior to the change to verify that it's actually changing in response to the orientation.